addon/validations/factory.js

```diff
@@ -189,6 +189,14 @@ function normalizeOptions(validations = {}, globalOptions = {}) {
   });
 }
 
+function classProperty(value, property) {
```

Comment:
BaseValidator requires this since it's imported and there for does not go through `factoryFor`.  This method knows how to deal with reading constructor props off types output from factoryFor and regular types.